pystring.cpp

```diff
@@ -1080,15 +1089,15 @@ namespace path
     void splitdrive_nt(std::string & drivespec, std::string & pathspec,
                        const std::string & p)
     {
-        if(pystring::slice(p, 1, 2) == ":")
+        if(pystring::slice(p, 1, 2) == colon)
```

Comment:
Doesn't this make more sense as

    if (p.size() >= 2 && p[1] == ':')

??? There's no need to use slice to create a new string that's one character long, just to compare it to another string that contains only a colon, right?

Reply:
Yes, that makes more sense, and runs faster.
I updated the MR.